Source/WebCore/rendering/svg/SVGResourcesCache.cpp

```diff
@@ -108,6 +108,8 @@ static inline bool rendererCanHaveResources(RenderObject& renderer)
 
 void SVGResourcesCache::clientStyleChanged(RenderElement& renderer, StyleDifference diff, const RenderStyle* oldStyle, const RenderStyle& newStyle)
 {
+    ASSERT(!renderer.element() || renderer.element()->isSVGElement());
```

Comment:
The first condition is also still suspicious :-) Do we call clientStyleChanged() for anonymous renderers?
I think it doesn't actually add much value to add this assertion here, no? We're no longer using renderer.element() in this function.

Reply:
ok, I need to debug a bit then since the condition is needed (at least for debug builds, otherwise there are crashes).